src/adapters/postgres/assigntenant-adapter.ts

```diff
@@ -0,0 +1,122 @@
+export class PostgresAssignTenantService {
+    constructor(
+        @InjectRepository(UserTenantMapping)
+        private userTenantMappingRepository: Repository<UserTenantMapping>,
+        @InjectRepository(User)
+        private userRepository: Repository<User>,
+        @InjectRepository(Tenants)
+        private tenantsRepository: Repository<Tenants>,
+    ) { }
+    public async createAssignTenant(request: any, createAssignTenantDto: CreateAssignTenantDto) {
+        try {
+
+            const userId = createAssignTenantDto.userId;
+            const tenantIds = createAssignTenantDto.tenantId;
+
+            // Check if tenant array is not empty
+            if (!tenantIds || tenantIds.length === 0) {
+                return new SuccessResponse({
+                    statusCode: HttpStatus.BAD_REQUEST,
+                    message: "Roles array cannot be empty.",
+                });
+            }
+            let result = [];
+            let errors = [];
+
+            for (const tenantId of tenantIds) {
+                // If role already exists for user, return error response
+                console.log(userId);
+                console.log(tenantId);
+
+                let findExistingRole = await this.userTenantMappingRepository.findOne({
+                    where: {
+                        userId: userId,
+                        tenantId: tenantId,
+                    },
+                });
+                if (findExistingRole) {
+                    errors.push({
+                        errorMessage: `User is already exist in ${tenantId} Tenant.`,
+                    });
+                    continue;
+                }
+
+                // User is exist in user table 
+                let userExist = await this.userRepository.findOne({
+                    where: {
+                        userId: userId,
+                    },
+                });
+                if (!userExist) {
+                    errors.push({
+                        errorMessage: `User ${userId} is not exist.`,
+                    });
+                    continue;
+                }
+
+                // User is exist in user table 
+                let tenantExist = await this.tenantsRepository.findOne({
+                    where: {
+                        tenantId: tenantId,
+                    },
+                });
+                if (!tenantExist) {
+                    errors.push({
+                        errorMessage: `Tenant ${tenantId} is not exist.`,
+                    });
+                    continue;
+                }
+
+
+                const data = await this.userTenantMappingRepository.save({
+                    userId: userId,
+                    tenantId: tenantId,
+                    createdBy: request['user'].userId,
+                    updatedBy: request['user'].userId
+                })
+
+                result.push(new ResponseAssignTenantDto(data, `Tenant assigned successfully to the user.`));
```

Comment:
result.push(new ResponseAssignTenantDto(data, `User is successfully added to the Tenants.`));